src/eddymotion/utils.py

```diff
@@ -0,0 +1,169 @@
+import random
+from itertools import chain, zip_longest
+from typing import Generator
+
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright 2024 The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+"""Iterators to traverse the volumes in a 4D image."""
+
+
```

Comment:
With Generator[int] instead of Generator[int,None,None] it returns this error "TypeError: Too few arguments for typing.Generator; actual 1, expected 3"

Reply:
You're right, as per the documentation: https://docs.python.org/3/library/typing.html#typing.Generator.

But, as the documentation also indicates at the bottom, we can instead annotate these as `Iterator[int]`

Reply:
(`Iterator[int]` seems more informative to me)